packages/compass-telemetry/src/ipc-track.ts

```diff
@@ -0,0 +1,31 @@
+import isElectronRenderer from 'is-electron-renderer';
+import type { HadronIpcRenderer } from 'hadron-ipc';
+import {
+  type TelemetryPreferences,
+  createGenericTrack,
+  type TrackFunction,
+} from './track';
+
+function emit(
+  ipc: HadronIpcRenderer | null | undefined,
+  event: string,
+  data: Record<string, any>
+): void {
+  // We use ipc.callQuiet instead of ipc.call because we already
+  // print debugging messages below
+  void ipc?.callQuiet?.(event, data);
+  if (typeof process !== 'undefined' && typeof process.emit === 'function') {
+    (process as any).emit(event, data);
+  }
+}
+
+export function createTrack(
+  preferencesService?: TelemetryPreferences
+): TrackFunction {
+  // This application may not be running in an Node.js/Electron context.
+  const ipc: HadronIpcRenderer | null | undefined = isElectronRenderer
```

Comment:
This is a bit redundant, hadron-ipc already does the check

Reply:
(still a thing, but a non-blocker, just FYI that we can have one less dependency in this package and don't need to use require import as the hadron-ipc will do the dynamic export for us)

Reply:
will do! with compass-web providing a different send method, this check should be superfluous anyway

Reply:
will do!

Reply:
I thought this was mainly about avoiding compass-web importing the electron code, but we've avoided that by providing a different send method